### PR TITLE
enable aiter gemm_a8w8_bpreshuffle for ptpc gemm

### PR DESCRIPTION
**Pull request by Yuechguo**
## Motivation
enable aiter gemm_a8w8_bpreshuffle for gemm in per-token-per-channel quant computing

## Modifications
1.Modified the weight preloading part to support weight shuffle.
2.enable aiter gemm_a8w8_bpreshuffle op as an alternative to torch._scaled_mm, it does not strictly depend on a specific version of PyTorch.

## Accuracy Test
`USE_ROCM_AITER_ROPE_BACKEND=0 \
SGLANG_USE_AITER=1 \
CUDA_VISIBLE_DEVICES=0 \
python3 -m sglang.launch_server \
        --model-path /pretrained_model/Qwen/Qwen3-32B-FP8-Dynamic \
        --served-model-name  Qwen3-32B-FP8-Dynamic \
        --attention-backend aiter \
        --host 127.0.0.1 \
        --port 28000 \
        --tp-size 1 --trust-remote-code \
        --max-prefill-tokens 32768 \
        --chunked-prefill-size 32768 \
        --context-len 32768 \
        --max-running-requests 512 \
        --disable-radix-cache --page-size 64 \`

TP=1：
| Qwen3-32B-FP8-Dynamic        |  gsm8k score |
| :--------------------------- | :------- | 
| origin（torch2.7.1+hipblaslt） |  0.90± 0.01|
| aiter  gemm_a8w8_bpreshuffle |  0.91 ± 0.01|

## Benchmark & Profiling
`SGLANG_USE_AITER=1 \
CUDA_VISIBLE_DEVICES=0,1 \
python3 -m sglang.launch_server \
        --model-path /pretrained_model/Qwen/Qwen3-32B-FP8-Dynamic \
        --served-model-name  Qwen3-32B-FP8-Dynamic \
        --attention-backend aiter \
        --host 127.0.0.1 \
        --port 28000 \
        --tp-size 2 --trust-remote-code \
        --max-prefill-tokens 32768 \
        --chunked-prefill-size 32768 \
        --context-len 32768 \
        --max-running-requests 512 \
        --disable-radix-cache --page-size 64 \`

TP=2:
| Qwen3-32B-FP8-Dynamic        | Mean TTFT | Mean ITL |
| :--------------------------- | :------- | :------- |
| origin（torch2.7.1+hipblaslt） |  2256.57  |    20.35 |
| aiter  gemm_a8w8_bpreshuffle |  2175.97  |    17.26 |

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/fp8_utils.py [hunk 1]
```diff
@@ -606,25 +609,35 @@ def apply_fp8_linear(
                 use_per_token_if_dynamic
                 and not per_tensor_weights
                 and not per_tensor_activations
-                and USE_ROWWISE_TORCH_SCALED_MM
+                and USE_HIP_SCALED_MM
```
Comment (HaiShaw) [inline python/sglang/srt/layers/quantization/fp8_utils.py]: break change - to case other than HIP, should keep original here.

File: python/sglang/srt/layers/quantization/fp8_utils.py [hunk 2]
```diff
@@ -56,22 +56,25 @@
...
         # The condition to determine if it is on a platform that supports
         # torch._scaled_mm rowwise feature.
         # The condition is determined once as the operations
         # are time consuming.
         return get_device_capability() >= (9, 4) and _TORCH_VERSION_TUPLE >= (2, 7, 0)
     return False
 
-
-USE_ROWWISE_TORCH_SCALED_MM = use_rowwise_torch_scaled_mm()
+USE_HIP_SCALED_MM = use_hip_scaled_mm()
```
Comment (HaiShaw) [inline python/sglang/srt/layers/quantization/fp8_utils.py]: try to simplify and clean this

File: python/sglang/srt/layers/quantization/fp8_utils.py [hunk 3]
```diff
@@ -56,22 +56,25 @@
...
     if _is_hip:
+        if _use_aiter:
+            return get_device_capability() >= (9, 4)
+        
+        _TORCH_VERSION = torch.__version__.split("+")[0]
+        try:
+            _TORCH_VERSION_TUPLE = tuple(map(int, _TORCH_VERSION.split(".")[:3]))
+        except ValueError:
+            _TORCH_VERSION_TUPLE = (0, 0, 0)
+
```
Comment (HaiShaw) [inline python/sglang/srt/layers/quantization/fp8_utils.py]: simplify and clean this, not to override original `use_rowwise_torch_scaled_mm`

File: python/sglang/srt/layers/quantization/compressed_tensors/schemes/compressed_tensors_w8a8_fp8.py [hunk 4]
```diff
@@ -76,7 +91,14 @@ def process_weights_after_loading(self, layer) -> None:
             else:
                 weight_scale = layer.weight_scale.data
 
-            layer.weight = Parameter(weight.t(), requires_grad=False)
+            if USE_AITER_SCALED_MM:
```
[the remark this answers is not in the document]
Comment (HaiShaw) [reply python/sglang/srt/layers/quantization/compressed_tensors/schemes/compressed_tensors_w8a8_fp8.py]: Take gemini comment here, try not to define functions and alias excessively, readability is low and run time cost is high

**Review by HaiShaw — CHANGES_REQUESTED** [review]

File: python/sglang/srt/layers/quantization/fp8_utils.py (hunk 1 above)
Comment (Yuechguo) [reply python/sglang/srt/layers/quantization/fp8_utils.py]: yeah，I will make minimal changes

File: python/sglang/srt/layers/quantization/fp8_utils.py [hunk 5]
```diff
@@ -606,25 +605,35 @@ def apply_fp8_linear(
...
-                    weight,
-                    out_dtype=input.dtype,
-                    scale_a=x_scale,
-                    scale_b=weight_scale.t(),
-                    bias=bias,
-                )
-                return _process_scaled_mm_output(output, input_2d.shape, output_shape)
-
+                if _use_aiter:                    
+                    output = gemm_a8w8_bpreshuffle(
```
Comment (HaiShaw) [inline python/sglang/srt/layers/quantization/fp8_utils.py:664]: add comments above this

**Comment by HaiShaw** [comment]
@Yuechguo perf and accuracy data?

**Review by HaiShaw — APPROVED** [review]
